bescort.lic

```diff
@@ -584,6 +590,64 @@ class Bescort
       end
     end
   end
+  
+  def take_theren_rope_bridge(mode)
+	theren_side_roomid = 8650
+	rossman_side_roomid = 8637
+	@target_room = false
+	@shuffle_direction = false
+    if mode == 'totheren'
+		if not Room.current.id == rossman_side_roomid
+			echo("Must start on the south side of the rope bridge")
+			exit
+		end
+		@shuffle_direction = "north"
+		@target_room = theren_side_roomid
+	elsif mode == 'torossman'
+		if not Room.current.id == theren_side_roomid
+			echo("Must start on the north side of the rope bridge")
+			exit
+		end
+		@shuffle_direction = "south"
+		@target_room = rossman_side_roomid
+	else
+		echo("Unrecognized argument: #{mode}")
+		exit
+	end
+	
+	start_room = Room.current.id
+	Flags.add("rope_wait","finally arriving on this side","finally reaching the far side")
+	while Room.current.id == start_room
+		put("retre") # RT irrelevant
+		put("retre")
+		Flags.reset("rope_wait") # avoiding race conditions
+		ret = bput("climb rope", "Roundtime", "is already on the rope", "You climb onto a", "You are engaged")
+		if ret.include?("is already on the rope")
+			sleep(0.1) until Flags["rope_wait"]
+			Flags.reset("rope_wait")
+		else
+			sleep(2)
+			waitrt?
+		end
+	end
+	
+	start_time = Time.now.to_f
+	until Room.current.id == @target_room or Time.now.to_f - start_time > 300
```

Comment:
As a general style and safety guideline we prefer `||` over `or` since they have different operator precedence. It's probably also worth parentheses around the time difference just to make precedence explicit.